core/src/main/java/dk/alexandra/fresco/framework/sce/SecureComputationEngineImpl.java

```diff
@@ -16,79 +17,98 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Secure Computation Engine - responsible for having the overview of things and setting everything
  * up, e.g., based on properties.
  */
-public class SecureComputationEngineImpl<ResourcePoolT extends ResourcePool, Builder extends ProtocolBuilder>
-    implements SecureComputationEngine<ResourcePoolT, Builder> {
+public class SecureComputationEngineImpl
+    <ResourcePoolT extends ResourcePool, BuilderT extends ProtocolBuilder>
+    implements SecureComputationEngine<ResourcePoolT, BuilderT> {
 
-  private ProtocolEvaluator<ResourcePoolT, Builder> evaluator;
+  private ProtocolEvaluator<ResourcePoolT, BuilderT> evaluator;
   private ExecutorService executorService;
   private boolean setup;
-  private ProtocolSuite<ResourcePoolT, Builder> protocolSuite;
+  private ProtocolSuite<ResourcePoolT, BuilderT> protocolSuite;
   private static final AtomicInteger threadCounter = new AtomicInteger(1);
-  private final static org.slf4j.Logger logger =
-      LoggerFactory.getLogger(SecureComputationEngineImpl.class);
+  private static final Logger logger = LoggerFactory.getLogger(SecureComputationEngineImpl.class);
+  private Duration timeOut = Duration.ofMinutes(10);
 
-  public SecureComputationEngineImpl(ProtocolSuite<ResourcePoolT, Builder> protocolSuite,
-      ProtocolEvaluator<ResourcePoolT, Builder> evaluator) {
+  /**
+   * Creates a new {@link SecureCompatationEngineImpl}.
+   *
+   * @param protocolSuite the {@link ProtocolSuite} to use to evaluate the secure computation
+   * @param evaluator the {@link ProtocolEvaluator} to run secure evaluation.
+   */
+  public SecureComputationEngineImpl(ProtocolSuite<ResourcePoolT, BuilderT> protocolSuite,
+      ProtocolEvaluator<ResourcePoolT, BuilderT> evaluator) {
     this.protocolSuite = protocolSuite;
     this.evaluator = evaluator;
     this.setup = false;
   }
 
+  /**
+   * Sets the time out duration for waiting for output from running applications.
+   *
+   * <p>
+   * Defaults to a value of 10 minutes.
+   * </p>
+   *
+   * @param timeOut the new time out duration
+   */
+  public void setRunTimeout(Duration timeOut) {
```

Comment:
Exposing this as a setter seems wrong.

It belongs in the construction - and is currently only exposed in order to do tests

Reply:
Previously the timeout arbitrarily set to 10 minutes and was unchangeable, which definitely seemed wrong. I dont I have strong opinion about whether it should be configured in the constructor or as a setter, I just think it should be possible for the user to define. Is it not desirable that one could change the timeout during the lifetime of a SCE object? If not, I guess the constructor is the way to go.

Reply:
Well

Either use start or run application, 
If you want control, use start - if just default behaviour, then run is fine.

My point is start is for the advanced usages - and no, you never need to change the timeout after the creation.

Reply:
How about passing it as a parameter to the runApplication method where it is used?

Reply:
Good idea, but leave a default implementation in the interface of SecureComputationEngine in order to keep a version of run and start similar

Reply:
How about this way then?